zerver/lib/rate_limiter.py

```diff
@@ -32,12 +32,14 @@ class RateLimitedObject(ABC):
     def __init__(self, backend: Optional['Type[RateLimiterBackend]']=None) -> None:
         if backend is not None:
             self.backend = backend  # type: Type[RateLimiterBackend]
+        elif settings.RUNNING_INSIDE_TORNADO:
+            self.backend = TornadoInMemoryRateLimiterBackend
```

Comment:
I basically agree with Anders.  There's two ways we could enforce this:
* Having code in `TornadoInMemoryRateLimiterBackend` to throw an `AssertionError` for rate-limiters other than the basic requests limiter for `GET /events`
* Changing this logic to select the backend based on the rule type

I'm not sure what's cleaner, but I think we want it to be a strong security structure that the local-to-process rate limiting can only be used for the basic API request counters relevant to Tornado, and not as a way to get `numprocesses` times our intended allocation of attempts to check a password (for example).

I also think it makes sense to do the expedient of merging this as-is and fixing those in the next week.  I'm OK with doing so because I'm confident the only rate-limiting counters accessible from Tornado code paths are fine today, and this hardening/assertion issue can be resolved as part of the bundle of follow-up cleanups that we're going to do afterwards.  And also because I think it'll be easier to do the conversations around that in a new PR thread that doesn't already have 45+ messages on it.

I certainly consider that work to be a release-blocker for including this PR in a release.